test/unit/org/apache/cassandra/cql3/ColumnSpecificationTest.java

```diff
@@ -0,0 +1,206 @@
+        verifyColumnSpec("name text CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+    }
+
+    @Test
+    public void testAlterTableAddColumnWithCheck()
+    {
+        createTable("CREATE TABLE %s (pk text primary key);");
+        execute("ALTER TABLE %s ADD name text CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+        verifyColumnSpec("name text CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+    }
+
+    @Test
+    public void testAlterTableAddColumnWithMask()
+    {
+        createTable("CREATE TABLE %s (pk text primary key);");
+        execute("ALTER TABLE %s ADD name text MASKED WITH system.mask_default()");
+        verifyColumnSpec("name text MASKED WITH system.mask_default()");
+    }
+
+    @Test
+    public void testAlterTableAddColumnWithMaskAndCheck()
+    {
+        createTable("CREATE TABLE %s (pk text primary key);");
+        execute("ALTER TABLE %s ADD name text MASKED WITH system.mask_default() CHECK NOT_NULL(name)");
+        verifyColumnSpec("name text MASKED WITH system.mask_default() CHECK NOT_NULL(name)");
+    }
+
+    @Test
+    public void testAlterTableAddColumnWithMaskAndMultipleChecks()
+    {
+        createTable("CREATE TABLE %s (pk text primary key);");
+        execute("ALTER TABLE %s ADD name text MASKED WITH system.mask_default() CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+        verifyColumnSpec("name text MASKED WITH system.mask_default() CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+    }
+
+    /**
+     * TODO - investigate if it is possible to specify checks before mask when creating a table
+     */
+    @Test(expected = RuntimeException.class)
+    public void testFailingCreateTableWithColumnHavingMaskAfterCheck()
+    {
+        createTable("CREATE TABLE %s (pk text primary key, name text CHECK NOT_NULL(name) AND LENGTH(name) > 1 MASKED WITH system.mask_default());");
+    }
+
+    /**
+     * TODO - investigate if it is possible to specify both check and mask, check being first
+     */
+    @Test(expected = RuntimeException.class)
+    public void testFailingAlterTableAlterColumnWithCheckAndMask()
+    {
+        createTable("CREATE TABLE %s (pk text, name text, primary key (pk));");
+        execute("ALTER TABLE %s ALTER name CHECK NOT_NULL(name) AND LENGTH(name) > 1 MASKED WITH system.mask_default();");
+        verifyColumnSpec("name text MASKED WITH system.mask_default() CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+    }
+
+    /**
+     * TODO - investigate if it is possible to specify both check and mask, mask being first
+     */
+    @Test(expected = RuntimeException.class)
+    public void testFailingAlterTableAlterColumnWithMaskAndCheck()
+    {
+        createTable("CREATE TABLE %s (pk text, name text, primary key (pk));");
+        execute("ALTER TABLE %s ALTER name MASKED WITH system.mask_default() CHECK NOT_NULL(name) AND LENGTH(name) > 1");
+        verifyColumnSpec("name text MASKED WITH system.mask_default() CHECK NOT_NULL(name) AND LENGTH(name) > 1");
```

Comment:
I don't think we are never reaching this `verifyColumnSpec` right? The runtime exception is thrown by the execute.

Reply:
same as below